### PR TITLE
Update lowering-module test case

### PR DESCRIPTION
**Pull request by rasendubi**
Both test annotation and test result changed.

### CODE REVIEW

File: src/traits/lowering-module.md [hunk 1]
```diff
@@ -45,19 +45,18 @@ compiler will then invoke the `program_clauses_for` query on that
...
+error: program clause dump
+  --> $DIR/lower_impl.rs:5:1
    |
-LL | #[rustc_dump_program_clauses] //~ ERROR Implemented(T: Foo) :-
+LL | #[rustc_dump_program_clauses]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-error: aborting due to previous error
+   |
+   = note: forall<T> { Implemented(T: Foo) :- ProjectionEq(<T as std::iter::Iterator>::Item == i32), TypeOutlives(T: 'static), Implemented(T: std::iter::Iterator), Implemented(T: std::marker::Sized). }
```
Comment (rasendubi) [inline src/traits/lowering-module.md]: There was a continuation of the following form before:
```
... TypeOutlives(T \
: 'static), ...
```

Not sure if that was intentional, but it did not render properly and produced `\` in the output.

I just let the line continue as it does in the actual code.

**Comment by rasendubi** [comment]
CI failure is not relevant to `lowering-module.md`:

```
There were 2 broken links:
traits/lowering-rules.md#31: "https://github.com/rust-lang/chalk/blob/master/chalk-rules/src/clauses.rs" returned 404 Not Found
traits/wf.md#14: "https://github.com/rust-lang/chalk/blob/master/chalk-rules/src/wf.rs" returned 404 Not Found
2019-04-20 17:27:17 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
2019-04-20 17:27:17 [ERROR] (mdbook::utils): Error: Rendering failed
2019-04-20 17:27:17 [ERROR] (mdbook::utils): 	Caused By: The "linkcheck" renderer failed
The command "mdbook build" exited with 101.
```

File: src/traits/lowering-module.md (hunk 1 above)
Comment (mark-i-m) [reply src/traits/lowering-module.md]: I think that should be ok. Thanks :)

**Comment by mark-i-m** [comment]
Yes, I will ignore the CI failures for the time being, as they are doing a lot of refactoring on chalk.

**Pull request merged**